site/viamguides/src/control-module/control-module.md

```diff
@@ -0,0 +1,577 @@
+    >       if has_person:
+    >           LOGGER.info("Person detected. Turning on lamp.")
+    >           await self.lamp.do_command({"toggle_on": []})
+    >           self.last_seen = now
+    >       else:
+    >           if not hasattr(self, "last_seen"):
+    >               self.last_seen = now
+    >           elapsed = now - self.last_seen
+    >           if elapsed > 180:
+    >               LOGGER.info("No person detected for 3 minutes. Turning off lamp.")
+    >               await self.lamp.do_command({"toggle_off": []})
+    >   except Exception as err:
+    >       LOGGER.error(f"Error in control logic: {err}")
+    >   await asyncio.sleep(10)
+    > ```
+
+1.  **Validate configuration**: In the `validate_config()` method, you can ensure the user correctly provides the required dependencies, formatted as expected. If your goal is a working version that validates nothing, proceed to the next step.
+
+    > aside positive
+    > **Optional**: In the example shown here, the validation requires `struct_to_dict`, so we've added it to our imports.
+    >
+    > ```python
+    > from viam.utils import struct_to_dict
+    > ```
+    >
+    > The user must configure a vision service and generic component. An error message is returned if the requirements are not provided as the correct data type.
+    >
+    > ```python
+    > @classmethod
+    > def validate_config(cls, config: ComponentConfig) -> Sequence[str]:
+    >    """This method allows you to validate the configuration object received from the machine, as well as to return any implicit dependencies based on that `config`.
+    >    Args:
+    >        config (ComponentConfig): The configuration for this resource
+    >    Returns:
+    >        Sequence[str]: A list of implicit dependencies
+    >    """
+    >    attrs = struct_to_dict(config.attributes)
+    >    required_dependencies = ["vision", "generic"]
+    >    implicit_dependencies = []
+    >
+    >    for component in required_dependencies:
+    >        if component not in attrs or not isinstance(attrs[component], str):
+    >            raise ValueError(f"{component} is required and must be a string")
+    >        else:
+    >            implicit_dependencies.append(attrs[component])
+    >     return implicit_dependencies
+    > ```
+
+1.  **Start the control loop and initialize other resources**: In the `reconfigure()` method, start the control logic to run in a background loop. This method is called when your model is first added to the machine, and again whenever the machine configuration is updated. If your control logic relies on other parts of the machine (like sensors or services), you should also initialize those resources here.
+    ```python
+    def reconfigure(
+      self, config: ComponentConfig,
+      dependencies: Mapping[ResourceName, ResourceBase]
+      ):
+      # starts automatically
+      if self.running is None:
+          self.start()
+      else:
```

Comment:
This should probably go after configuring the dependencies, right? Maybe swap this step and the next one.